src/Pos/Generator/Block/Mode.hs

```diff
@@ -357,9 +350,13 @@ instance MonadBlockGenBase m => MonadBListener (BlockGenMode m) where
     onApplyBlocks = onApplyBlocksStub
     onRollbackBlocks = onRollbackBlocksStub
 
-instance MonadBlockGenBase m => MonadDiscovery (BlockGenMode m) where
-    getPeers = getPeersSum
-    findPeers = findPeersSum
+instance ( Monad m, MonadKnownPeers m ) => MonadKnownPeers (BlockGenMode m) where
+    updateKnownPeers = lift . updateKnownPeers
+    addKnownPeers = lift . addKnownPeers
+    removeKnownPeer = lift . removeKnownPeer
+
+instance ( Monad m, MonadFormatPeers m ) => MonadFormatPeers (BlockGenMode m) where
+    formatKnownPeers formatter = lift (formatKnownPeers formatter)
```

Comment:
Instead of defining such instances in many places we usually define an overlappable instance next to the class definition.